SU2_CFD/include/integration/CIntegrationFactory.hpp

```diff
@@ -0,0 +1,60 @@
+/*!
+ * \file CIntegrationFactory.hpp
+ * \brief Headers of the CIntegrationFactory class
+ * \author T. Albring
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "../../include/solvers/CSolverFactory.hpp"
+
+class CIntegration;
+class CConfig;
+
+class CIntegrationFactory{
+
+private:
+
+  /*!
+   * \brief Private constructor to avoid creating instances of this class
+   */
+  CIntegrationFactory();
+
+public:
+
+  /*!
+   * \brief Create the integration container based on the current main solver
+   * \param[in] kindSolver    - The kind of main solver
+   * \param[in] solver        - The solver container
+   * \return                  - Pointer to the allocated integration container
+   */
+  static CIntegration** createIntegrationContainer(ENUM_MAIN_SOLVER kindSolver, CSolver **solver_container);
```

Comment:
const const solver_container please.

Ok, why not just delegate the numerics creation to the solvers then? That's pretty much what it boils down to, and it would be one step in the direction of moving the numerics into the solvers.

Reply:
I am gonna create a separate PR to deal with the numerics.